etl/migrations/20251127090000_schema_change_messages.sql

```diff
@@ -0,0 +1,173 @@
+-- Schema change logical messages (DDL)
+-- Adds helpers and trigger to emit logical decoding messages when tables change.
+
+create or replace function etl.describe_table_schema(
+    p_table oid
+) returns table (
+    name text,
+    type_oid oid,
+    type_name text,
+    type_modifier int,
+    ordinal_position int,
+    primary_key_ordinal_position int,
+    nullable boolean
+)
+language plpgsql
+stable
+as
+$$
+declare
+    v_query text;
+begin
+    v_query := format(
+        'with direct_parent as (
+            select i.inhparent as parent_oid
+            from pg_inherits i
+            where i.inhrelid = %1$s
+            limit 1
+        ),
+        primary_key as (
+            select x.attnum, x.n as position
+            from pg_constraint con
+            join unnest(con.conkey) with ordinality as x(attnum, n) on true
+            where con.contype = ''p''
+              and con.conrelid = %1$s
+        ),
+        parent_primary_key as (
+            select a.attname, x.n as position
+            from pg_constraint con
+            join unnest(con.conkey) with ordinality as x(attnum, n) on true
+            join pg_attribute a on a.attrelid = con.conrelid and a.attnum = x.attnum
+            join direct_parent dp on dp.parent_oid = con.conrelid
+            where con.contype = ''p''
+        )
+        select
+            a.attname::text,
+            a.atttypid,
+            case
+                when tn.nspname = ''pg_catalog'' then t.typname
+                else tn.nspname || ''.'' || t.typname
+            end::text,
+            a.atttypmod,
+            a.attnum::int,
+            coalesce(pk.position, ppk.position)::int,
+            not a.attnotnull
+        from pg_attribute a
+                 join pg_type t on t.oid = a.atttypid
+                 join pg_namespace tn on tn.oid = t.typnamespace
+                 left join primary_key pk on pk.attnum = a.attnum
+                 left join parent_primary_key ppk on ppk.attname = a.attname
+        where a.attrelid = %1$s
+          and a.attnum > 0
+          and not a.attisdropped
+          and a.attgenerated = ''''
+        order by a.attnum;',
+        p_table
+    );
+
+    return query execute v_query;
+end;
+$$;
+
+create or replace function etl.emit_schema_change_messages()
+returns event_trigger
+language plpgsql
+as
+$$
+declare
+    cmd record;
+    table_schema text;
+    table_name text;
+    table_oid oid;
+    schema_json jsonb;
+    msg_json jsonb;
+    v_wal_level text;
+begin
+    -- Check if logical replication is enabled; if not, silently skip.
+    -- This prevents crashes when Supabase ETL is installed but wal_level != logical.
+    v_wal_level := current_setting('wal_level', true);
+    if v_wal_level is null or v_wal_level != 'logical' then
```

Comment:
Nit: `v_wal_level != 'logical'` should be enough. (`NULL != 'logical'` is `true`). 

Can `wal_level` be `null`?